common/src/test/java/org.neo4j.spark.service/AuthenticationTest.java

```diff
@@ -0,0 +1,60 @@
+package org.neo4j.spark.service;
+
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.neo4j.driver.AuthToken;
+import org.neo4j.driver.Config;
+import org.neo4j.driver.GraphDatabase;
+import org.neo4j.spark.util.DriverCache;
+import org.neo4j.spark.util.Neo4jDriverOptions;
+import org.neo4j.spark.util.Neo4jOptions;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.sparkproject.guava.io.BaseEncoding;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.ArgumentMatchers.*;
+import static org.mockito.Mockito.times;
+
+@PrepareForTest({GraphDatabase.class})
+@RunWith(PowerMockRunner.class)
+public class AuthenticationTest {
+
+    @Test
+    public void testLdapConnectionToken() {
```

Comment:
@utnaf can we write this in scala?

Reply:
Nope, can get it working in Scala, sadly there is an issue with power mock that doesn't make it work

Reply:
Jk, it was the other test I tried to do that wasn't working in scala, this one is. Will move it to the scala package.